.github/workflows/teams.yml

```diff
@@ -0,0 +1,54 @@
+# A workflow to run the generate_teams.py script in the CI weekly and create a pull request
+
+on:
+  schedule:
+    # Run every Monday at 03:00 AM UTC
+    - cron: "0 3 * * 1"
+  workflow_dispatch:
+
+jobs:
+  generate-teams:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout main branch of repository
+        uses: actions/checkout@v3
+        with:
+          ref: main
+          submodules: recursive
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - run: python scripts/generate_teams.py
+
+      - name: Run the teams generation script
+        run: python scripts/generate_teams.py
+
+      - name: Run pre-commit hooks
+        run: |
+          pip install pre-commit
+          pre-commit install
```

Comment:
This should not be required if you're running `pre-commit --all-files`

Reply:
Makes sense. I will remove `pre-commit install`, though `pre-commit` isn't installed by default, or is it?

Reply:
"install" here is confusing haha. The pip installation should be done as you do. `pre-commit install` is required to execute `pre-commit` before every commit, that is, it installs `pre-commit` as a git hook. You can still use `pre-commit` without installing it as a git hook (which is being done here).

Reply:
I see, so the install step is redundant and we can do `pre-commit run` directly. Removing it